### PR TITLE
[query] Actually allow the service to use the shuffler

### PR DESCRIPTION
**Pull request by johnc1231**

### CODE REVIEW

**Review by tpoterba — APPROVED** [review]

**Review by danking — DISMISSED** [review]

**Comment by johnc1231** [comment]
@tpoterba Enough has changed that I think this should have a compiler team review check on my fixes to Shuffling.

File: hail/src/main/scala/is/hail/expr/ir/lowering/LowerDistributedSort.scala [hunk 1]
```diff
@@ -120,31 +121,47 @@ object LowerDistributedSort {
...
+
+    val initialInterval = Interval(intervalRange(0), intervalRange(1), true, true)
+    val initialSegment = SegmentResult(IndexedSeq(0), initialInterval, initialChunks)
 
     val totalNumberOfRows = initialChunks.map(_.size).sum
-    val idealNumberOfRowsPerPart = Math.max(1, totalNumberOfRows / inputStage.numPartitions)
+    optTargetNumPartitions.foreach(i => assert(i >= 1, s"Must request positive number of partitions. Requested ${i}"))
+    val targetNumPartitions = optTargetNumPartitions.getOrElse(inputStage.numPartitions)
+
+    val idealNumberOfRowsPerPart = if (targetNumPartitions == 0) { 1 } else {
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/expr/ir/lowering/LowerDistributedSort.scala:144]: style nit: no braces necessary around 1

File: hail/src/test/scala/is/hail/expr/ir/lowering/LowerDistributedSortSuite.scala [hunk 2]
```diff
@@ -72,7 +72,7 @@ class LowerDistributedSortSuite extends HailSuite {
   }
 
   @Test def testDistributedSort(): Unit = {
-    val tableRange = TableRange(100, 10)
+    val tableRange = TableRange(40, 10)
```
Comment (tpoterba) [inline hail/src/test/scala/is/hail/expr/ir/lowering/LowerDistributedSortSuite.scala:75]: revert?

**Review by tpoterba — CHANGES_REQUESTED** [review]
minor things

File: hail/src/test/scala/is/hail/expr/ir/lowering/LowerDistributedSortSuite.scala (hunk 2 above)
Comment (johnc1231) [reply hail/src/test/scala/is/hail/expr/ir/lowering/LowerDistributedSortSuite.scala:75]: Oh yeah, might as well revert

**Review by danking — APPROVED** [review]

**Pull request merged**